test/integration/pod_shutdown_test.go

```diff
@@ -140,6 +148,24 @@ func shutdownAllPodsOfApp(
 	// pod should have given up the lease during its graceful shutdown.
 	waitForLeaderElectionLeaseToHaveHolderIdentity(t, namespace, appName,
 		func(holder string) bool { return holder == "" }, 1*time.Minute)
+
+	if immediateRestart {
```

Comment:
If any of the above assertions fail, then this code will not be reached, and the cluster will be left in a bad state for all other integration tests. The t.Cleanup should be adjusted to ensure that the pods are conditionally restarted if thy have no already been restarted even when immediateRestart is true.

Reply:
Yeah this part was meant to be just some temporary code while I could get the test working.

Reply:
I've broken out a separate function that performs an immediate pod restart.